components/browser/engine-system/src/main/java/mozilla/components/browser/engine/system/window/SystemWindowRequest.kt

```diff
@@ -0,0 +1,44 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.browser.engine.system.window
+
+import android.os.Message
+import android.webkit.WebView
+import mozilla.components.browser.engine.system.SystemEngineSession
+import mozilla.components.concept.engine.EngineSession
+import mozilla.components.concept.engine.window.WindowRequest
+
+/**
+ * WebView-based implementation of [WindowRequest].
+ *
+ * @property webView the WebView from which the request originated.
+ * @property newWebView the WebView to use for opening a new window, may be null for close requests.
+ * @property openAsDialog whether or not the window should be opened as a dialog, defaults to false.
+ * @property triggeredByUser whether or not the request was triggered by the user, defaults to false.
+ * @property resultMsg the message to send to the new WebView, may be null.
+ */
+class SystemWindowRequest(
+    private val webView: WebView,
+    private val newWebView: WebView? = null,
+    val openAsDialog: Boolean = false,
+    val triggeredByUser: Boolean = false,
```

Comment:
Currently we do not use `triggeredByUser`, right? I wonder whether we should think about this in a follow-up. `WindowRequest`s that are not triggered by any user action is something we probably want to block (popups). It would be interesting to know whether GeckoView does that by default; or whether we would want such a flag from them.

Reply:
I think GV blocks by default, right? In Webview, we can control this using the `javaScriptCanOpenWindowsAutomatically` setting it seems. So, yes, I guess a follow-up to decide how we want to handle this. I will file one.

Reply:
Yeah, I talked to snorp. GV blocks by default. There's a strong indication that we should just drop everything where `triggeredByUser` is false.

Reply:
Made a note of it in the follow-up: https://github.com/mozilla-mobile/android-components/issues/1503